pyproject.toml

```diff
@@ -3,7 +3,6 @@ build-backend = "setuptools.build_meta"
 requires = [
     "setuptools>=64.0.0",
     "tomli  ; python_version < '3.11'",
-    "versioneer>=0.24",
 ]
 
 [project]
```

Comment:
Since `version` was removed from `setup.py`, I believe it needs to be added here instead (eg https://github.com/rapidsai/cudf/blob/branch-23.08/python/dask_cudf/pyproject.toml#L12). And I think the `dynamic` field should be removed.

Also, an entry needs to be added to `update-version.sh` to update this file's version too.

You can see that no version is being used in the wheel build: https://github.com/rapidsai/dask-cuda/actions/runs/5377443612/jobs/9755962220?pr=1204#step:4:54

Reply:
Thanks @raydouglass for catching that. I've done the changes in https://github.com/rapidsai/dask-cuda/pull/1204/commits/210937e76ce282e687f83e2e79e72397878a3424 and it looks like the build version is correct now: https://github.com/rapidsai/dask-cuda/actions/runs/5379525987/jobs/9760908314#step:4:54 .

We have `RAPIDS_DATE_STRING` being passed to the package version in [`meta.yaml`](https://github.com/rapidsai/dask-cuda/blob/branch-23.08/conda/recipes/dask-cuda/meta.yaml#L20), which seems consistent with cuDF. Is there anywhere else we need to pass it?

Reply:
I don't know if this is acceptable, but I've attempted something in https://github.com/rapidsai/dask-cuda/pull/1204/commits/937919be6a757f63880b017fa460873c95d31aac . That change will essentially change `version` in `pyproject.toml` before building the wheel and later revert it. I don't know how we would identify whether we are doing a release build (and thus not modify `pyproject.toml`) or a pre-release build that we want to do that change, any ideas?

Obviously, we would also need to update `ci/release/update-version.sh` to do the necessary changes in this file if we go ahead with that solution, but want to check first whether this is how we want to move ahead.

cc @vyasr who may have insights here.

Reply:
Inside this script you have access to a bash function that you can use: `if rapids-is-release-build; do ...; done`

Reply:
Nice, you guys thought of everything! 😄